packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -27,67 +33,161 @@ import {
+    showTimeZone: boolean;
+    showTSeparator: boolean;
+    timeZone: string;
+  };
+}
+
+interface FormattingSectionContentState {
+  formatSettings?: FormatterItem[];
+  formatRulesChanged: boolean;
+  showTimeZone?: boolean;
+  showTSeparator?: boolean;
+  timeZone?: string;
+  defaultDateTimeFormat?: string;
+  defaultDecimalFormatOptions?: FormatOption;
+  defaultIntegerFormatOptions?: FormatOption;
+  truncateNumbersWithPound?: boolean;
+  timestampAtMenuOpen: Date;
+}
+
+export class FormattingSectionContent extends PureComponent<
+  FormattingSectionContentProps,
+  FormattingSectionContentState
+> {
+  static defaultProps = {
+    scrollTo: (): void => undefined,
+    defaults: {
+      defaultDateTimeFormat:
+        DateTimeColumnFormatter.DEFAULT_DATETIME_FORMAT_STRING,
+      defaultDecimalFormatOptions: {
+        defaultFormatString: DecimalColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      defaultIntegerFormatOptions: {
+        defaultFormatString: IntegerColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      showTimeZone: false,
+      showTSeparator: true,
+      timeZone: DateTimeColumnFormatter.DEFAULT_TIME_ZONE_ID,
+    },
+  };
+
   static inputDebounceTime = 250;
 
-  static focusFirstInputInContainer(container) {
-    const input = container.querySelector('input, select, textarea');
+  static focusFirstInputInContainer(container: HTMLElement | null): void {
+    const input = container?.querySelector('input, select, textarea');
     if (input) {
-      input.focus();
+      (input as HTMLElement).focus();
     }
   }
 
-  static isSameDecimalOptions(options1, options2) {
+  static isSameOptions(
+    options1?: FormatOption,
+    options2?: FormatOption
+  ): boolean {
     return options1?.defaultFormatString === options2?.defaultFormatString;
   }
 
-  static isSameIntegerOptions(options1, options2) {
-    return options1?.defaultFormatString === options2?.defaultFormatString;
+  static isSameDecimalOptions(
+    options1?: FormatOption,
+    options2?: FormatOption
+  ): boolean {
+    return FormattingSectionContent.isSameOptions(options1, options2);
+  }
+
+  static isSameIntegerOptions(
+    options1?: FormatOption,
+    options2?: FormatOption
+  ): boolean {
+    return FormattingSectionContent.isSameOptions(options1, options2);
   }
 
-  static isValidColumnName(name) {
-    return name && DbNameValidator.isValidColumnName(name);
+  static isValidColumnName(name: string): boolean {
+    return (
+      name != null && name !== '' && DbNameValidator.isValidColumnName(name)
+    );
   }
 
-  static isValidFormat(columnType, format) {
+  static isValidFormat(
+    columnType: string,
+    format: Partial<TableColumnFormat>
+  ): boolean {
     // Undefined or empty string formats are always invalid
     if (!columnType || !format.formatString) {
       return false;
     }
 
+    const { formatString } = format;
+    assertNotNull(formatString);
+
+    const nullCheckedFormat = { formatString };
```

Comment:
Instead, create a predicate to check that formatString isn't null, e.g.:
```
isFormatStringFormat(format: Partial<TableColumnFormat>): format is Pick<TableColumnFormat, 'formatString'>
```

Then you don't need to create a new object each time you check